### PR TITLE
Datasource: Pass access type to data source instance settings

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
As a part of data source backend migration (for Elastic, Prometheus, ...) we need to know on frontend, if data source uses browser/server access mode. In this PR we are passing information about access to frontend

More https://docs.google.com/document/d/1CzmOgShzXmtac9Tcq5CT8-4_q3n1i2Kd88nZDEZCLGw and https://github.com/grafana/grafana/pull/31579/files#r594475244
**Which issue(s) this PR fixes**:

### CODE REVIEW

File: packages/grafana-data/src/types/datasource.ts [hunk 1]
```diff
@@ -599,6 +599,7 @@ export interface DataSourceInstanceSettings<T extends DataSourceJsonData = DataS
   password?: string; // when access is direct, for some legacy datasources
   database?: string;
   isDefault?: boolean;
+  access?: string;
```
Comment (Elfo404) [inline packages/grafana-data/src/types/datasource.ts:602]: would it make sense in your opinion to have `'direct' | 'server'` instead?
also in https://github.com/grafana/grafana/blob/32a551b4caae9ac6c6c4ba3d006d7096a9023a02/packages/grafana-data/src/types/datasource.ts#L568

also, I think this is always set (at least according to the types above), so what do you think about
```suggestion
  access: string;
```
?

**Review by Elfo404 — APPROVED** [review]
looks good! 🔥 

left a comment regarding the types, but not a blocker. Also I'm not sure if `'direct' | 'server'` ar the only valid values for `access`

File: packages/grafana-data/src/types/datasource.ts (hunk 1 above)
Comment (ivanahuckova) [reply packages/grafana-data/src/types/datasource.ts]: Yeah you are right! I got slightly uncertain as the url is also optional, but it is non-optional in  https://github.com/grafana/grafana/blob/32a551b4caae9ac6c6c4ba3d006d7096a9023a02/packages/grafana-data/src/types/datasource.ts#L569 .

**Comment by ivanahuckova** [comment]
I was thinking about this as well and I am not sure. It should be only `direct` or `proxy` -https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx#L19 - but everywhere it is typed as string. I mean we can type it as direct and proxy, as that's what it is.

**Comment by Elfo404** [comment]
> I was thinking about this as well and I am not sure. It should be only `direct` or `proxy` -https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx#L19 - but everywhere it is typed as string. I mean we can type it as direct and proxy, as that's what it is.

i think string is fine as well, we can also make it strictier later on if we need it 👍

**Comment by ivanahuckova** [comment]
>i think string is fine as well, we can also make it strictier later on if we need it 👍

I was actually thinking about it, and I think you were right - we should do `direct | proxy` because if we don't type it strongly now, we probably won't do it anymore. We can always turn it into string, if we need. 🙂

**Pull request merged**